src/ngraph/op/extractimagepatches.cpp

```diff
@@ -0,0 +1,157 @@
+
+// ExtractImagePatches v3
+
+constexpr NodeTypeInfo op::v3::ExtractImagePatches::type_info;
+
+op::v3::ExtractImagePatches::ExtractImagePatches(const Output<Node>& image,
+                                                 const Shape& sizes,
+                                                 const Strides& strides,
+                                                 const Shape& rates,
+                                                 const PadType& padding)
+    : Op({image})
+    , m_patch_sizes(sizes)
+    , m_patch_movement_strides(strides)
+    , m_patch_selection_rates(rates)
+    , m_padding(padding)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v3::ExtractImagePatches::validate_and_infer_types()
+{
+    const PartialShape input_Pshape = get_input_partial_shape(0);
+
+    NODE_VALIDATION_CHECK(this,
+                          get_input_element_type(0).is_integral_number(),
+                          "input tensor must be an integral number.");
+    NODE_VALIDATION_CHECK(this, input_Pshape.rank() == 4, "input tensor must be 4D tensor.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_sizes.size() == 2,
+                          "Attribute sizes should be in [size_rows, size_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_movement_strides.size() == 2,
+                          "Attribute strides should be in [stride_rows, stride_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_movement_strides[0] > 0 && m_patch_movement_strides[1] > 0,
+                          "Attribute strides should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_selection_rates.size() == 2,
+                          "Attribute rates should be in [rate_rows, rate_cols] format.");
+
+    NODE_VALIDATION_CHECK(this,
+                          m_patch_selection_rates[0] > 0 && m_patch_selection_rates[1] > 0,
+                          "Attribute rates should be strictly greater than zeros in values.");
+
+    NODE_VALIDATION_CHECK(
+        this,
+        m_padding == PadType::VALID || m_padding == PadType::SAME_LOWER ||
+            m_padding == PadType::SAME_UPPER,
+        "Attribute padding should be in either valid or same_lower or same_upper.");
+
+    if (input_Pshape.is_dynamic())
+    {
+        set_input_is_relevant_to_shape(0);
+        auto output_Pshape = PartialShape::dynamic(4);
+        set_output_type(0, get_input_element_type(0), output_Pshape);
+    }
+    else
+    {
+        Shape input_shape = get_input_shape(0); // as input shape is static
+        size_t out_rows(0);
+        size_t out_cols(0);
+
+        if (m_patch_movement_strides[0] > input_shape[2] ||
+            m_patch_movement_strides[1] > input_shape[3])
+        {
+            if (m_padding == PadType::VALID)
+            {
+                out_rows = 0; // no padding allowed for VALID
```

Comment:
I don't quite understand why the out_rows and out_cols are 0 in this case. If the rates value is bigger than the corresponding input dimension then the first element will still have index 0 in input so it will be in the output patch. The rest of the input elements will be in position m_patch_movement_strides[0] * i > input_shape[2] so they will not be in the output patch. So we will have out_rows = out_cols = 1.

Reply:
Thanks for showing me the bug. I solve the issues. 
You are right. In case of larger stride, the actual equation will handle this scenario. However, there might a corner case when rates are very larger than the input dimension, in that case there is a possibility my out_rows and out_cols become negative in padding::valid case (the Padding::valid should give (0,0) and same_lower and same_upper should result (1,1) in this situation)
I fixed this by using signed int for out_rows and out_cols and later check it's negativity. Before adding this to output_shape, it was type_cast-ed to size_t.